lib/GADS/Schema/Result/File.pm

```diff
@@ -183,6 +189,26 @@ sub export_hash
 # Created by DBIx::Class::Schema::Loader v0.07043 @ 2015-11-13 16:02:57
 # DO NOT MODIFY THIS OR ANYTHING ABOVE! md5sum:266q8eJmPiCjcNhwddaUkw
 
+sub _build_recordsource { 'File'; }
+sub _build_valuefield { undef; }
+
+# I've done this to override the role - I don't know how else to best implement this!
+sub purge {
+    my $self = shift;
+
+    my $source = $self->recordsource or error __"No recordsource defined";
+    my $schema = $self->result_source->schema;
+
+    $schema->txn_do(sub {
+      my $file = $schema->resultset($source)->search({},{prefetch=>'value'})->find($self->id);
+      my $value = $file->value;
+      $value->update({
```

Comment:
I think this could simply be `$self->update` and remove the previous 2 lines?

Reply:
I feel I should add context to why I do this - other languages I've previously worked with don't allow this, so it's an old habit, I shall try to remember